bemani/backend/gf/gfv8.py

```diff
@@ -0,0 +1,2657 @@
+        battledata.add_child(Node.u32('win', 0))
+        battledata.add_child(Node.u32('lose', 0))
+        battledata.add_child(Node.u8('score_type', 0))
+        battledata.add_child(Node.s16('strategy_item', 0))
+        battledata.add_child(Node.s16('production_item', 0))
+        battledata.add_child(Node.u32('draw', 0))
+        battledata.add_child(Node.u8('max_class', 0))
+        battledata.add_child(Node.u16('max_rensyo', 0))
+        battledata.add_child(Node.u16('vip_rensyo', 0))
+        battledata.add_child(Node.s32('max_defeat_skill', 0))
+        battledata.add_child(Node.s32('max_defeat_battle_rate', 0))
+        battledata.add_child(Node.u32('gold_star', 0))
+        battledata.add_child(Node.u32('random_select', 0))
+        battledata.add_child(Node.u8('enable_bonus_bp', 0))
+        battledata.add_child(Node.u32('type_normal', 0))
+        battledata.add_child(Node.u32('type_perfect', 0))
+        battledata.add_child(Node.u32('type_combo', 0))
+
+        battle_aniv = Node.void('battle_aniv')
+        battledata.add_child(battle_aniv)
+
+        get = Node.void('get')
+        battle_aniv.add_child(get)
+
+        category_ver = Node.u16_array (
+            'category_ver',
+            [
+                0, 0,
+                0, 0,
+                0, 0,
+                0, 0,
+                0, 0,
+                0,
+            ]
+        )
+
+        get.add_child(category_ver)
+        category_genre = Node.u16_array (
+            'category_genre',
+            [
+                0, 0,
+                0, 0,
+                0, 0,
+                0, 0,
+                0, 0,
+                0,
+            ]
+        )
+        get.add_child(category_genre)
+
+        battledata.add_child(Node.u8_array('area_id_list', [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,]))
+        battledata.add_child(Node.u32_array('area_win_list', [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,]))
+        battledata.add_child(Node.u32_array('area_lose_list', [ 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,]))
+
+        battledata.add_child(Node.u32('perfect', profile.get_int('perfect', 0)))
+        battledata.add_child(Node.u32('great', profile.get_int('great', 0)))
+        battledata.add_child(Node.u32('good', profile.get_int('good', 0)))
+        battledata.add_child(Node.u32('poor', profile.get_int('poor', 0)))
+        battledata.add_child(Node.u32('miss', profile.get_int('miss', 0)))
+
+        history = Node.void('history')
+        battledata.add_child(history)
+
+        round_before = Node.void('round')
```

Comment:
Again with the completely unpopulated chunks, why is this not handled and not in a loop? You have access to the score history, this doesn't look done at all.